src/components/ChemPage.tsx

```diff
@@ -28,37 +59,40 @@ export default function ChemPage(props: { result: Molecule, setChem: Dispatch<Se
                         structure={props.result.smiles}
                         width={350}
                         height={300}
-                        svgMode
+                        svgMode={true}
                     />
                 </section>
             </div>
             <div className="glass-container active p-3 mt-4">
                 <section>
-                    <h3>Molecule Details</h3>
+                    <div className='flex flex-row justify-between items-center'>
+                        <h3>Molecule Details</h3>
+                        <DownloadLink/>
+                    </div>
                     <table className="chem-details-table">
                         <tbody>
                         <tr>
-                            <th className="font-bold">MOA:</th>
+                            <DetailsHeading label='MOA' source='IDG'/>
```

Comment:
In the future, we'll need to actually fetch the source information from the API/DB. But this is okay for now as a placeholder.